### PR TITLE
prevent non numerical input

### PR DESCRIPTION
**Pull request by hcvcastro**
- browser: add 'getLang' function to get the current ...
- browser: use Intl built-in object to get number format
- browser: improve _preventNonNumericalInput function

### CODE REVIEW

File: browser/src/control/Control.JSDialogBuilder.js [hunk 1]
```diff
@@ -1518,6 +1542,10 @@ L.Control.JSDialogBuilder = L.Control.extend({
 	_metricfieldControl: function(parentContainer, data, builder, customCallback) {
 		var value;
 		var controls = builder._controlHandlers['basespinfield'](parentContainer, data, builder, customCallback);
+		if (!L.Browser.chrome) {
+			controls.spinfield.onkeypress = L.bind(builder._preventNonNumericalInput, builder);
```
Comment (eszkadev) [inline browser/src/control/Control.JSDialogBuilder.js:1546]: can't we add this only one time in `basespinfield` which is shared in metric and formatted field?

File: browser/src/control/Control.JSDialogBuilder.js [hunk 2]
```diff
@@ -285,6 +292,20 @@ L.Control.JSDialogBuilder = L.Control.extend({
...
 	},
 
+	_preventNonNumericalInput: function(e) {
+		e = e || window.event;
+		var charCode = (typeof e.which == 'undefined') ? e.keyCode : e.which;
+		var charStr = String.fromCharCode(charCode);
+		if (charStr.match(/^[0-9]+$/))
+			return;
+
+		for (var part in this._numberParts) {
```
Comment (eszkadev) [inline browser/src/control/Control.JSDialogBuilder.js:302]: this is not enough, we can still type:
'1.1,,,,,,' or '-13-----'

Best would be to use single regexp where we can use minus sign only on the beginning and separators at correct positions. This regex should check input field value, not only current key code

File: browser/src/control/Control.JSDialogBuilder.js (hunk 1 above)
Comment (hcvcastro) [reply browser/src/control/Control.JSDialogBuilder.js:1546]: are we sure that basespinfield class is numeric?
what about integers, currency?

File: browser/src/control/Control.JSDialogBuilder.js (hunk 2 above)
Comment (hcvcastro) [reply browser/src/control/Control.JSDialogBuilder.js:302]: oh, sound good, thanks

**Comment by eszkadev** [comment]
now it looks better :)
I think we can add also `onchange` event listener which will remove '-' signs which are not at the first position ? what do you think?

File: browser/js/global.js [hunk 3]
```diff
@@ -228,7 +228,13 @@ window.app = {
 
 		// @property retina: Boolean
 		// `true` for browsers on a high-resolution "retina" screen.
-		retina: (window.devicePixelRatio || (window.screen.deviceXDPI / window.screen.logicalXDPI)) > 1
+		retina: (window.devicePixelRatio || (window.screen.deviceXDPI / window.screen.logicalXDPI)) > 1,
+
+		getLang: function () {
+			return (navigator.languages && navigator.languages.length ? navigator.languages[0] :
+				navigator.language || navigator.userLanguage ||
+				navigator.browserLanguage || navigator.systemLanguage);
```
Comment (Ashod) [inline browser/js/global.js:236]: Is this supposed to be 
`navigator.languages && navigator.languages.length ? navigator.languages[0] : (navigator.language || navigator.userLanguage ||
				navigator.browserLanguage || navigator.systemLanguage)` 
or 
`(navigator.languages && navigator.languages.length ? navigator.languages[0] :
				navigator.language) || navigator.userLanguage ||
				navigator.browserLanguage || navigator.systemLanguage` ?

Perhaps adding `()` would make it more readable.

**Review by Ashod — APPROVED** [review]
Thanks @hcvcastro. One comment, but otherwise looks good.

**Comment by hcvcastro** [comment]
Thanks, I hope it is clear now

**Pull request merged**